assets/src/components/form/validations/index.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * External imports
+ */
+import { __, _n, sprintf } from '@eventespresso/i18n';
+
+/**
+ * @function
+ * @param {mixed} value
+ * @return {string|null} error string on fail
+ */
+export const isFloat = value => {
+	// first convert to string
+	value = value + '';
+	// convert decimal to known format
+	value = value.replace( ',', '.' );
+	// strip all trailing zeros
+	value = value.replace( /0+$/, '' );
+	const lastChar = value.substr( value.length - 1 );
+	const float = (
+		// parsed value is same as original coerced to a number
+		parseFloat( value, 10 ) === +value &&
+		(
+			// no remainder but last character is a decimal
+			// ie: original value like 1.0
+			( value % 1 === 0 && lastChar === '.' ) ||
+			// has a remainder and last character is NOT a decimal
+			// ie: original value like 1.# where # is not 0
+			( value % 1 !== 0 && lastChar !== '.' )
+		)
+	);
+	return ! float ? __( 'Value must be a float', 'event_espresso' ) : null;
```

Comment:
In my opinion `isFloat` should just return a boolean value.  This implementation here appears to be a specific sanitization process or validation with error (likely for the forms system).  So I'd do something like:

```
// isFloat and its verification logic would be moved as I mentioned in an above comment
import { isFloat } from `@eventespresso/helpers';
import { __ } from '@eventespresso/i18n';

const verifyFloatForForm = ( value ) => ! isFloat( value ) ? __( 'Value must be a float', 'event_espresso' ) : null;
```

*Note* the comment here can be extrapolated for all the other "validators" in this file.